packages/block-editor/src/components/block-preview/index.js

```diff
@@ -92,6 +93,17 @@ export function BlockPreview( {
  */
 export default memo( BlockPreview );
 
+function InnerStyles() {
+	const { styles } = useSelect( ( select ) => {
+		const settings = select( blockEditorStore ).getSettings();
+		return {
+			styles: settings.styles,
+		};
+	}, [] );
+
+	return <EditorStyles styles={ styles } />;
```

Comment:
How are these styles being added right now, or previously before the bug?
Why does EditorStyles not include these already?

Reply:
Prior to the bug being introduced the styles were output via a portal, so even in a child preview with its own block editor provider, the generated styles were being output by the parent. However, since moving over to the block editor store-based state approach for style overrides, this preview no-longer propagates (or renders) any styles generated by the blocks within its preview, because the local-to-the-preview state isn't being used anywhere. Or to put it differently `setStyleOverride` will only update the preview's block editor store, and not the parent environment's block editor store, which is where `EditorStyles` is currently used.

The proposed fix in this PR is to ensure that the preview also outputs `EditorStyles`, but only those styles generated from within the preview.

I've updated the PR description with the following:

Prior to this PR, for things that use useBlockPreview, the styles that are generated within that preview were not output anywhere. This is because ExperimentalBlockEditorProvider uses its own registry, so is designed to have values passed down to it, but not to propagate values outside of itself. So any calls to setStyleOverride would update state within the local block editor provider, and not be captured by the parent's EditorStyles.